src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -817,6 +818,13 @@ namespace ScatterKernelDetail{
 
       template < typename T_DeviceAllocator >
       static void* initHeap( T_DeviceAllocator* heap, void* pool, size_t memsize){
+        if( pool == NULL && memsize != 0 )
+        {
+          throw std::invalid_argument(
+            "Scatter policy cannot use NULL for non-empty memory pools.\
```

Comment:
Wow, I was actually not aware of the correct way to split a string in C++. Thanks for your input! :sparkling_heart: